backend/core/src/secondary-adapters/postgres/video/video-repository-adapter.ts

```diff
@@ -412,32 +412,53 @@ export class VideoRepositoryAdapter implements VideoRepository {
     });
   }
 
-  searchByCategories({ skip, take, categories }: CategorySearchRequestQueryDto): Promise<VideoWithChannel[]> {
-    return this.prismaClient.video.findMany({
-      where: {
-        ...{ privacy: StreamPrivacy.PUBLIC },
-        categories: {
-          some: {
-            category: {
-              name: {
-                in: categories,
+  async searchByCategories({
+    skip,
+    take,
+    categories,
+  }: CategorySearchRequestQueryDto): Promise<{ list: VideoWithChannel[]; total: number }> {
```

Comment:
i think, better be pointed response-dto for Promise<> instead current solution

Reply:
done